cmake/libxmp-checks.cmake

```diff
@@ -17,10 +17,6 @@ elseif(CMAKE_COMPILER_IS_GNUCC)
     set(HAVE_GCC TRUE)
 endif()
 
-if(HAVE_CLANG AND NOT WIN32 AND NOT CMAKE_LINKER) # Workaround
-    find_program(CMAKE_LINKER ld)
-endif()
-
```

Comment:
I don't know cmake internals good. We should have a way of reporting
_only the first line_ of linker output, killing anything beginning with `\n`
and/or `\r`, i.e. the linker and then its version signature.

Reply:
The problem is not because the linker reports the wrong output, the problem is that it just wasn't found at all, otherwise, if there is an invalid linker binary, it will fail differently, I bet.

Reply:
> The problem is not because the linker reports the wrong output, the problem is that it just wasn't found at all, otherwise, if there is an invalid linker binary, it will fail differently, I bet.

Yes, I know.

What I said was something general, i.e.: cmake already does what I say
for the compiler, so why don't we make it do the same for the linker (of
course when it finds one..)

Reply:
This is a case when I made the fake "ld" executable that prints the swearing word and always returns code 1, the CC test got failed immediately:
![изображение](https://user-images.githubusercontent.com/6751442/127748694-402c1e0f-b6a7-4bc1-ae58-6ae78b7c8c51.png)

However, when I passed the `CC=clang-11`, it successfully ignored the fake linker and used real:
![изображение](https://user-images.githubusercontent.com/6751442/127748742-b4706b1c-fe57-40be-9b23-4d128f8b6116.png)

But, what happens if I enforce it to use the fake linker:
![изображение](https://user-images.githubusercontent.com/6751442/127748817-4e5f2d98-6912-4010-9f43-112293743888.png)
HA! It works even with a fake linker

Reply:
However, I made the a dirty trick and CLang configure got broken immediately:
- I renamed the real "ld" into "ld-bak" to make CLang detect the fake linker
- I ran the configure, and clang successfully failed the link because it used the fake linker:
![изображение](https://user-images.githubusercontent.com/6751442/127748909-15ad51fa-5e45-4dc4-a6e9-8c9b52c54045.png)

Reply:
What I mean is, some thing like this:
```diff
diff --git a/cmake/libxmp-checks.cmake b/cmake/libxmp-checks.cmake
index d1eaad1..bff9d48 100644
--- a/cmake/libxmp-checks.cmake
+++ b/cmake/libxmp-checks.cmake
@@ -23,3 +23,3 @@ execute_process(COMMAND ${CMAKE_LINKER} -v
         ERROR_VARIABLE  LINKER_OUTPUT)
-message(STATUS "${LINKER_OUTPUT}")
+message(STATUS "Linker identification: ${LINKER_OUTPUT}")
 if("${LINKER_OUTPUT}" MATCHES ".*GNU.*" OR "${LINKER_OUTPUT}" MATCHES ".*with BFD.*")
```

Say, when I invoke cmake with `-DCMAKE_LINKER=/opt/cctools-port/bin/i686-apple-darwin9-ld`
I get:
```
-- Check for working linker: /opt/cctools-port/bin/i686-apple-darwin9-ld
-- Linker identification: 241.9
configured to support archs: armv4t armv5 armv6 armv7 armv7f armv7k armv7s armv6m armv7m armv7em armv8 arm64 arm64v8 i386 x86_64 x86_64h
LTO support using: LLVM version 3.4.2

```

See the multi-lines? Can we lose everything beginning with the newline?
I.e. something like we do in C is:
```c
char *p = strchr(str,'\n'); if(p) *p = 0;
char *p = strchr(str,'\r'); if(p) *p = 0;
```
Is there some equivalent in cmake string() functions?

Reply:
Lemme try some...

Reply:
Just now I made the fake LD that prints multiline:
![Снимок экрана от 2021-07-31 21-54-15](https://user-images.githubusercontent.com/6751442/127749834-299b4ab4-54f4-45c0-a912-7b771bdc190e.png)
and it works fine at me

Reply:
It works fine even I use \r\n way (in this example, the \r\n\n result is)
![изображение](https://user-images.githubusercontent.com/6751442/127749858-30df8405-3c09-4038-bf84-545d7418242d.png)

Reply:
You can try yourself:
```
cd build
CC=clang cmake -DCMAKE_LINKER=$PWD/kold ..
```
[koko.tar.gz](https://github.com/libxmp/libxmp/files/6911718/koko.tar.gz)

Reply:
@sezero , what do you think on this?

Reply:
It is exactly the opposite of what I asked. (And never mind, that's the least of my worries.)

Reply:
I re-checked posts earlier, and, do you want to just print the first line only from the output?
that's ultra-easy:
![изображение](https://user-images.githubusercontent.com/6751442/127751106-a446340b-4948-4d68-b0e8-de019fa87844.png)

```cmake
string(REGEX REPLACE "[\r\n].*" "" LINKER_OUTPUT_LINE "${LINKER_OUTPUT}")
message(STATUS "Linker identification: ${LINKER_OUTPUT_LINE}")
```

Reply:
It works.  You can store the modified string in the same old LINKER_OUTPUT variable, no need for a new LINKER_OUTPUT_LINE

Reply:
> It works. You can store the modified string in the same old LINKER_OUTPUT variable, no need for a new LINKER_OUTPUT_LINE

I made the new variable with a goal to make the next check being not harmed in any random unexpected case

Reply:
I don't think that would ever happen, but OK.